cmd/bucket-replication.go

```diff
@@ -920,36 +913,46 @@ func (p *ReplicationPool) ResizeFailedWorkers(n int) {
 	}
 }
 
-func (p *ReplicationPool) queueReplicaTask(ctx context.Context, ri ReplicateObjectInfo) {
+func (p *ReplicationPool) queueReplicaFailedTask(ri ReplicateObjectInfo) {
 	if p == nil {
 		return
 	}
 	select {
-	case <-ctx.Done():
+	case <-GlobalContext.Done():
 		p.once.Do(func() {
 			close(p.replicaCh)
 			close(p.mrfReplicaCh)
```

Comment:
no need for ` 			close(p.replicaCh)` here

Reply:
@poornas it is done to handle both channels at the same time when GlobalContext is done.

Reply:
otherwise I would need two sync.Once @poornas

Reply:
this would lead to double close and panic - since sync.Once is called here to close both channels in queueReplicaFailedTask  **and** queueReplicaTask . queueReplicaFailedTask no longer processes replicaCh, so should really not be closing it

Reply:
nvm, understood what you wrote. this is ok